### DISCUSSION TITLE
Vectorstore with a kor library (on top of langchain)

### DISCUSSION THREAD

**Discussion by ahmed-bhs** [eyurtsev/kor#237] [Q&A] [answered]
I am testing a kor library and I find it really helpful when extracting pieces of information from a low-quality text (like parsed online ads). The problem is a kor prompt usually gets long and when combined with an original text, it easily exceeds the token limit of OpenAI.

When working with pure LangChain, I use vectorstores to grab the most relevant chunks of text.

Do you know how to use vectorstores with kor? That would really help! Thanks.

**Comment by eyurtsev** [comment, marked as answer]
Hi @ahmed-bhs,

There's no pre-built solution. You can iterate over all documents chunks as shown here: https://eyurtsev.github.io/kor/document_extraction.html

If you want to do something more efficient (but potentially more error prone), here's a discussion that might be useful: https://github.com/eyurtsev/kor/discussions/181
